File: app/src/main/java/in/completecourse/helper/PrefManager.java
package in.completecourse.helper;

import android.content.Context;
import android.content.SharedPreferences;

public class PrefManager {

    private final SharedPreferences pref;
    private SharedPreferences.Editor editor;

    // Shared preferences file name
    private static final String PREF_NAME = "vidya_complete_course";

    private static final String IS_FIRST_TIME_LAUNCH = "IsFirstTimeLaunch";
    private static final String IS_LOGGED_IN = "isLoggedIn";
    private static final String USER_CLASS = "user_class";

    public PrefManager(Context context) {
        // shared pref mode
        int PRIVATE_MODE = 0;
        this.pref = context.getSharedPreferences(PREF_NAME, PRIVATE_MODE);
    }

    public void setFirstTimeLaunch(boolean isFirstTime) {
        editor = pref.edit();
        editor.putBoolean(IS_FIRST_TIME_LAUNCH, isFirstTime);
        editor.apply();
    }

    public boolean isFirstTimeLaunch() {
        return pref.getBoolean(IS_FIRST_TIME_LAUNCH, true);
    }

    public void setLogin(boolean isLoggedIn) {
        editor.putBoolean(IS_LOGGED_IN, isLoggedIn);
        // commit changes
        editor.commit();
    }

    public boolean isLoggedIn(){
        return pref.getBoolean(IS_LOGGED_IN, false);
    }

    public void setUserClass(String userClass){
        editor.putString(USER_CLASS, userClass);
        editor.commit();
    }

    public String getUserClass(){
        return pref.getString(USER_CLASS, "");
    }
}

File: app/src/main/java/in/completecourse/helper/PrefManager.kt
package `in`.completecourse.helper

import `in`.completecourse.LoginActivity
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences

class PrefManager(context: Context) {
    private val pref: SharedPreferences
    private var editor: SharedPreferences.Editor? = null

    fun setFirstTimeLaunch(isFirstTime: Boolean) {
        editor = pref.edit()
        editor!!.putBoolean(IS_FIRST_TIME_LAUNCH, isFirstTime)
        editor?.apply()
    }

    fun isFirstTimeLaunch(): Boolean {
        return pref.getBoolean(IS_FIRST_TIME_LAUNCH, true)
    }

    fun saveUser(name: String?, userId: String?){
        editor = pref.edit()
        editor!!.putString(USERNAME, name)
        editor!!.putString(USERID, userId)
        editor?.apply()
    }

    fun getUserId(): String?{
        return pref.getString(USERID, "")
    }

    // --Commented out by Inspection START (2/11/19 12:17 AM):
    //    public String getUserClass(){
    //        return pref.getString(USER_CLASS, "");
    //    }
    // --Commented out by Inspection STOP (2/11/19 12:17 AM)

    fun logoutUser(context: Context) {
        editor = pref.edit()
        // Clearing all data from Shared Preferences
        editor?.clear()
        editor?.apply()

        // After logout redirect user to Login Activity
        val i = Intent(context, LoginActivity::class.java)
        // Closing all the Activities
        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)

        // Staring Login Activity
        context.startActivity(i)
    }

    companion object {
        // Shared preferences file name
        private const val PREF_NAME = "vidya_complete_course"
        private const val IS_FIRST_TIME_LAUNCH = "IsFirstTimeLaunch"
        private const val USERNAME = "username"
        private const val USERID = "userid"
    }

    init {
        // shared pref mode
        val privateMode = 0
        pref = context.getSharedPreferences(PREF_NAME, privateMode)
    }

}